neo/bin/prompt.py

```diff
@@ -55,6 +55,47 @@
 FILENAME_PROMPT_HISTORY = os.path.join(PATH_USER_DATA, '.prompt.py.history')
 
 
+class NeoHistory(FileHistory):
+    contract_meta_count = 0
+
+    def append(self, string):
+        string = self.redact_command(string)
+        if len(string) == 0:
+            return
+        self.strings.append(string)
+
+        # Save to file.
+        with open(self.filename, 'ab') as f:
+            def write(t):
+                f.write(t.encode('utf-8'))
+
+            write('\n# %s\n' % datetime.datetime.now())
+            for line in string.split('\n'):
+                write('+%s\n' % line)
+
+    def redact_command(self, string):
+        if len(string) == 0:
+            return string
+        command = [comm for comm in ['import wif', 'export wif', 'import nep2', 'export nep2'] if comm in string]
+        if len(command) > 0:
+            command = command[0]
+            if command in string and len(command + " ") < len(string):
+                return command + " <" + command.split(" ")[1] + ">"
+            else:
+                return string
+        elif "import contract" in string:
```

Comment:
The meta-count stuff needs documentation / comments, perhaps a function docstring. What are you trying achieve here?

Reply:
The `import contract` command has Metadata asking for name, version author, email and description that has to be removed. Since there are 5 metadata entries, the `contract_meta_count` variable keeps a count of how many metadata entries have taken place after the command `import contract`. Since we already know that there are going to be five entries after the `import contract` command, we just keep a count and not add them until all five are entered.

I will add documentation ASAP.

Reply:
Oh yeah, I hate if these names are part of the history, very good catch!

The problem here is that it wouldn't work if the users presses ctrl+c in one of these steps. I feel the proper solution there would be to not use prompt-toolkit to enter these 5 parameters, but a simple raw_input. Then we could get rid of all these steps here.

What do you think?

Reply:
Yeah, I think its definitely better idea. I wasn't too comfortable with not using prompt-toolkit, since its used everywhere. I'll do that.